app/controllers/users_controller.rb

```diff
@@ -1,36 +1,31 @@
 class UsersController < ApplicationController
-  before_action :if_user_is_logged_in, only: [:new ,:create]
+  before_action :if_user_is_logged_in, only: [ :new , :create]
+  before_action :if_user_is_admin, only: [:index]
+  before_action :if_user_is_manager, only: [:team_members]
+  before_action :logged_in_user, only: [:index, :team_members]
+  def create
+    user = User.from_omniauth(request.env["omniauth.auth"])
+    log_in user
 
 
 
-  def create
-    user = User.from_omniauth(env["omniauth.auth"])
-    log_in user
     redirect_to reviews_path
   end
 
   def index
-
     @users = User.all
-
   end
 
-
   def destroy
     log_out if logged_in?
     redirect_to root_path
-
   end
 
-
-
   def team_leads
     users = params[:managers].keys
     managers = params[:managers].values;
     User.update_all(manager: false)
-
     i = 0
-
     users.each do|user_id|
       User.where(id: managers[i]).update_all(manager: true)
```

Comment:
@chandu1307  I guess only one user will present for each managers values, So, you can use just update instead of update_all